src/views/VideoView/VideoView.tsx

```diff
@@ -70,6 +82,30 @@ const VideoView: React.FC<RouteComponentProps> = () => {
     })
   }, [addVideoView, videoID])
 
+  // Save the video timestamp
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const handleTimeUpdate = useCallback(
+    debounce((time) => {
+      if (data?.video?.id) {
+        updateWatchedVideos('INTERRUPTED', data.video.id, time)
+      }
+    }, 5000),
+    [data?.video?.id]
+  )
+
+  const handleVideoEnd = () => {
```

Comment:
Now I think of it, we probably should put all of those handlers in `useCallback` (also `handlePlay` and `handlePause`). Otherwise the event handlers will be set/unset on every render.